File: QUEUE/src/QueueOperations.java
public class QueueOperations{

    int[] Queue=new int[5];
    int front=-1,rear=-1;

    public void enqueue(int item){
        if(front==-1){
            front=0;
        }
        if(rear==Queue.length-1){

            System.out.println("Full");
            return;
        }
        else{
            Queue[++rear]=item;
            System.out.println("Added->"+item);
        }
    }
    public void dequeue(){
        if(front==-1||front>rear){

            System.out.println("Queue is empty");
            return;
        }
        else{
            System.out.println("Deleted ->"+Queue[front]);
            front++;
        }
    }
    public void display(){
        if(front==-1||front>rear){
            System.out.println("Queue is empty");
            return;
        }
        for(int i=front;i<=rear;i++){
            System.out.print(Queue[i]+"->");
        }
        System.out.println();
    }


    public static void main(String args[]){
        QueueOperations obj=new QueueOperations();
        obj.display();
        obj.enqueue(5);
        obj.enqueue(3);
        obj.enqueue(7);
        obj.display();
        obj.dequeue();
        obj.display();
        obj.enqueue(9);
        obj.enqueue(6);
        obj.enqueue(1);
        obj.display();
        obj.dequeue();
        obj.display();

    }
}


//see this also

// Queue implementation in Java

//public class Queue {
//    int SIZE = 5;
//    int items[] = new int[SIZE];
//    int front, rear;
//
//    Queue() {
//        front = -1;
//        rear = -1;
//    }
//
//    boolean isFull() {
//        if (rear == SIZE - 1) {
//            return true;
//        }
//        return false;
//    }
//
//    boolean isEmpty() {
//        if (front == -1)
//            return true;
//        else
//            return false;
//    }
//
//    void enQueue(int element) {
//        if (isFull()) {
//            System.out.println("Queue is full");
//        } else {
//            if (front == -1)
//                front = 0;
//            rear++;
//            items[rear] = element;
//            System.out.println("Inserted " + element);
//        }
//    }
//
//    int deQueue() {
//        int element;
//        if (isEmpty()) {
//            System.out.println("Queue is empty");
//            return (-1);
//        } else {
//            element = items[front];
//            if (front >= rear) {
//                front = -1;
//                rear = -1;
//            } /* Q has only one element, so we reset the queue after deleting it. */
//            else {
//                front++;
//            }
//            System.out.println("Deleted -> " + element);
//            return (element);
//        }
//    }
//
//    void display() {
//        /* Function to display elements of Queue */
//        int i;
//        if (isEmpty()) {
//            System.out.println("Empty Queue");
//        } else {
//            System.out.println("\nFront index-> " + front);
//            System.out.println("Items -> ");
//            for (i = front; i <= rear; i++)
//                System.out.print(items[i] + "  ");
//
//            System.out.println("\nRear index-> " + rear);
//        }
//    }
//
//    public static void main(String[] args) {
//        Queue q = new Queue();
//
//        // deQueue is not possible on empty queue
//        q.deQueue();
//
//        // enQueue 5 elements
//        q.enQueue(1);
//        q.enQueue(2);
//        q.enQueue(3);
//        q.enQueue(4);
//        q.enQueue(5);
//
//        // 6th element can't be added to because the queue is full
//        q.enQueue(6);
//
//        q.display();
//
//        // deQueue removes element entered first i.e. 1
//        q.deQueue();
//
//        // Now we have just 4 elements
//        q.display();
//
//    }
//}

File: QUEUE/src/py/Queueoperations.py
class CircularQueue:
    def __init__(self, capacity):
        self.capacity = capacity
        self.queue = [None] * capacity
        self.front = -1
        self.rear = -1

    # Check if queue is empty
    def isEmpty(self):
        return self.front == -1

    # Check if queue is full
    def isFull(self):
        return (self.rear + 1) % self.capacity == self.front

    # Add element to queue
    def enqueue(self, value):
        if self.isFull():
            print("Queue Overflow! Cannot enqueue", value)
        else:
            if self.front == -1:  # first element
                self.front = 0
            self.rear = (self.rear + 1) % self.capacity
            self.queue[self.rear] = value
            print(f"Enqueued {value}")

    # Remove element from queue
    def dequeue(self):
        if self.isEmpty():
            print("Queue Underflow! Cannot dequeue")
            return None
        value = self.queue[self.front]
        if self.front == self.rear:  # only one element
            self.front = self.rear = -1
        else:
            self.front = (self.front + 1) % self.capacity
        return value

    # Peek front element
    def peek(self):
        if self.isEmpty():
            print("Queue is empty")
            return None
        return self.queue[self.front]

    # Print current queue elements
    def print_queue(self):
        # print(self.queue)
        if self.isEmpty():
            print("Queue is empty")
            return
        print("Queue:", end=" ")
        i = self.front
        while True:
            print(self.queue[i], end=" ")
            if i == self.rear:
                break
            i = (i + 1) % self.capacity
        print()


# Example usage
q = CircularQueue(5)
q.enqueue(10)
q.enqueue(20)
q.enqueue(30)
q.enqueue(40)
q.enqueue(50)   # full now
q.print_queue()

print("Dequeued:", q.dequeue())
print("Dequeued:", q.dequeue())
q.print_queue()

q.enqueue(60)   # now reuses slot at front
q.enqueue(70)
q.print_queue()

print("Front element:", q.peek())



# class Queue:
    def __init__(self):
        self.q = []

    # Add element to the queue (rear side)
    def enqueue(self, ele):
        self.q.append(ele)
        print(f"Enqueued: {ele}")

    # Remove element from the queue (front side)
    def dequeue(self):
        if len(self.q) == 0:
            print("Queue is empty! Cannot dequeue.")
            return None
        removed = self.q.pop(0)  # remove first element
        print(f"Dequeued: {removed}")
        return removed

    # Print the current queue
    def print_q(self):
        if len(self.q) == 0:
            print("Queue is empty.")
        else:
            print("Current Queue:", self.q)


# Example usage
obj = Queue()
obj.enqueue(2)
obj.enqueue(64)
obj.enqueue(7)
obj.enqueue(1)

obj.print_q()   # Output: [2, 64, 7, 1]

obj.dequeue()   # Removes 2
obj.print_q()   # Output: [64, 7, 1]
